java/src/main/kotlin/io/spine/validation/java/DistinctGenerator.kt

```diff
@@ -51,19 +50,28 @@ internal class DistinctGenerator(ctx: GenerationContext) : SimpleRuleGenerator(c
      * the original collection is equal to the size of the `ImmutableSet` created as a copy
      * of the checked collection.
      */
-    override fun condition(): Expression {
-        val map = ctx.simpleRuleField.isMap
-        val fieldValue = ctx.fieldOrElement!!
-        val collectionToCount = if (map) MethodCall(fieldValue, "values") else fieldValue
+    override fun condition(): Expression<Boolean> {
+        val values = fieldValues()
         return equalityOf(
-            MethodCall(fieldValue, "size"),
+            MethodCall(values, "size"),
             ClassName(ImmutableSet::class)
-                .call("copyOf", listOf(collectionToCount))
+                .call<ImmutableSet<*>>("copyOf", values)
                 .chain("size")
         )
     }
 
-    private fun equalityOf(left: Expression, right: Expression): Expression {
-        return Literal(left.toCode() + " == " + right.toCode())
+    private fun fieldValues(): Expression<Collection<*>> {
+        val fieldIsMap = ctx.simpleRuleField.isMap
+        val fieldValue = ctx.fieldOrElement!!
+        return if (fieldIsMap) {
+            MethodCall(fieldValue, "values")
+        } else {
+            // We know that a `distinct` field is either a map or a `repeated` field.
```

Comment:
Please elaborate more on this. How do we know? Maybe, a reference to the code which guarantees that?

In any case, I wouldn't be so sure that here it's 100% so.